tapdb/multiverse_cache.go

```diff
@@ -13,6 +13,59 @@ import (
 	"github.com/lightningnetwork/lnd/lnutils"
 )
 
+// MultiverseCacheConfig is the configuration for the different multiverse
+// caches that exist.
+//
+//nolint:lll
+type MultiverseCacheConfig struct {
+	// ProofsPerUniverse is the number of proofs that are cached per
+	// universe. This number needs to be multiplied by the total number of
+	// universes to get the total number of proofs that are cached. There is
+	// no limit to the number of universes that can hold cached keys, so a
+	// cache is created for each universe that receives a request.
+	ProofsPerUniverse uint64 `long:"proofs-per-universe" description:"The number of proofs that are cached per universe."`
+
+	// LeavesNumCachedUniverses is the number of universes that can have a
+	// cache of leaf keys. Each cached universe can have up to
+	// LeavesPerUniverse keys cached. The total number of cached keys is
+	// therefore LeavesNumCachedUniverses * LeavesPerUniverse.
+	LeavesNumCachedUniverses uint64 `long:"leaves-num-cached-universes" description:"The number of universes that can have a cache of leaf keys."`
+
+	// LeavesPerUniverse is the number of leaf keys that are cached per
+	// universe. This number needs to be multiplied by
+	// LeavesNumCachedUniverses to get the total number of leaf keys that
+	// are cached.
+	LeavesPerUniverse uint64 `long:"leaves-per-universe" description:"The number of leaf keys that are cached per cached universe."`
+
+	// SyncerCacheEnabled is a flag that indicates if the syncer cache is
+	// enabled. The syncer cache is used to cache the set of active root
+	// nodes for the multiverse tree, which is specifically kept for the
+	// universe sync.
+	SyncerCacheEnabled bool `long:"syncer-cache-enabled" description:"If the syncer cache is enabled."`
+
+	// SyncerCachePreAllocSize is the pre-allocated size of the syncer
+	// cache.
+	SyncerCachePreAllocSize uint64 `long:"syncer-cache-pre-alloc-size" description:"The pre-allocated size of the syncer cache."`
+
+	// RootNodePageCacheSize is the size of the root node page cache that
+	// serves all paginated queries for root nodes that use different
+	// parameters than the syncer cache.
+	RootNodePageCacheSize uint64 `long:"root-node-page-cache-size" description:"The size of the root node page cache for all requests that aren't served by the syncer cache."`
+}
+
+// DefaultMultiverseCacheConfig returns the default configuration for the
+// multiverse cache.
+func DefaultMultiverseCacheConfig() MultiverseCacheConfig {
+	return MultiverseCacheConfig{
+		ProofsPerUniverse:        5,
```

Comment:
Temp for this commit? 

We should inherit the defaults defined in the existing package here.

Reply:
I think the original value for this was way too high. Because this is the number of proofs we cache **per universe** (and we don't have a limit on the number of universes we cache proofs for, it's a `lnutils.SyncMap` here and not an `lru.Cache`), we should be quite conservative here.

Perhaps we should turn this into an `lru.Cache` too, same as with the `universeLeafPageCache`?

Reply:
Ah ok, the original intent was for that to be a instance wide value (eg: only cache 50k proofs total). 

I think before merge, we should deploy a version of this on testnet/mainnet to observe the cache hit/miss rate.